code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -13,481 +13,374 @@ What are the archived variables for?
-		var/self_heat_capacity = heat_capacity(ARCHIVE)
-		sharer_heat_capacity = sharer_heat_capacity || sharer.heat_capacity(ARCHIVE)
-
-		if((sharer_heat_capacity > MINIMUM_HEAT_CAPACITY) && (self_heat_capacity > MINIMUM_HEAT_CAPACITY))
-			var/heat = conduction_coefficient*temperature_delta* \
-				(self_heat_capacity*sharer_heat_capacity/(self_heat_capacity+sharer_heat_capacity))
-
-			temperature = max(temperature - heat/self_heat_capacity, TCMB)
-			sharer_temperature = max(sharer_temperature + heat/sharer_heat_capacity, TCMB)
-			if(sharer)
-				sharer.temperature = sharer_temperature
-				if (initial(sharer.gc_share))
-					sharer.garbage_collect()
-	return sharer_temperature
-	//thermal energy of the system (self and sharer) is unchanged
-
-	///Compares sample to self to see if within acceptable ranges that group processing may be enabled
-	///Returns: a string indicating what check failed, or "" if check passes
-/datum/gas_mixture/proc/compare(datum/gas_mixture/sample)
-	var/list/sample_gases = sample.gases //accessing datum vars is slower than proc vars
-	var/list/cached_gases = gases
-
-	for(var/id in cached_gases | sample_gases) // compare gases from either mixture
-		var/gas_moles = cached_gases[id]
-		gas_moles = gas_moles ? gas_moles[MOLES] : 0
-		var/sample_moles = sample_gases[id]
-		sample_moles = sample_moles ? sample_moles[MOLES] : 0
-		var/delta = abs(gas_moles - sample_moles)
-		if(delta > MINIMUM_MOLES_DELTA_TO_MOVE && \
-			delta > gas_moles * MINIMUM_AIR_RATIO_TO_MOVE)
-			return id
-
-	var/our_moles
-	TOTAL_MOLES(cached_gases, our_moles)
-	if(our_moles > MINIMUM_MOLES_DELTA_TO_MOVE)
-		var/temp = temperature
-		var/sample_temp = sample.temperature
-
-		var/temperature_delta = abs(temp - sample_temp)
-		if(temperature_delta > MINIMUM_TEMPERATURE_DELTA_TO_SUSPEND)
-			return "temp"
-
-	return ""
-
-	///Performs various reactions such as combustion or fusion (LOL)
-	///Returns: 1 if any reaction took place; 0 otherwise
-/datum/gas_mixture/proc/react(datum/holder)
+/*
+/datum/gas_mixture/react(datum/holder)
 	. = NO_REACTION
-	var/list/cached_gases = gases
-	if(!length(cached_gases))
+	if(!total_moles())
 		return
 	var/list/reactions = list()
 	for(var/datum/gas_reaction/G in SSair.gas_reactions)
-		if(cached_gases[G.major_gas])
+		if(get_moles(G.major_gas))
 			reactions += G
 	if(!length(reactions))
 		return
 	reaction_results = new
-	var/temp = temperature
-	var/ener = THERMAL_ENERGY(src)
+	var/temp = return_temperature()
+	var/ener = thermal_energy()
 
 	reaction_loop:
 		for(var/r in reactions)
 			var/datum/gas_reaction/reaction = r
 
 			var/list/min_reqs = reaction.min_requirements
-			if(	(min_reqs["TEMP"] && temp < min_reqs["TEMP"]) || \
-				(min_reqs["ENER"] && ener < min_reqs["ENER"]) || \
-				(min_reqs["MAX_TEMP"] && temp > min_reqs["MAX_TEMP"])
-			)
+			if((min_reqs["TEMP"] && temp < min_reqs["TEMP"]) \
+			|| (min_reqs["ENER"] && ener < min_reqs["ENER"]))
```

Comment:
More "what is the purpose of this comment, + Don't change the min_recs check if this will ever see any use, it's needed for some of ghils reactions